wgsl/index.bs

```diff
@@ -8244,18 +8244,29 @@ Generally speaking, from the language's point-of-view, attributes can be
 ignored for the purposes of type and semantic checking.
 Additionally, the attribute name is a [=context-dependent name=], and
 some attribute parameters are also context-dependent names.
+Attributes in WGSL have the following general form, and each attribute defines further
+constraints describing the only valid forms:
 
-Unless explicitly permitted below, an attribute [=shader-creation error|must not=] be specified more than once per object or type.
+<pre class=include>
```

Comment:
This is false, because it shows the expression list form only.
The recent change made some operands be non-expressions.

It would be better to show the full `syntax/attribute` expansion here, or to give something that isn't an actual grammar rule here.
The latter could be either a prose description, or a schematic like we have in the variables and value declarations section.
See https://github.com/gpuweb/gpuweb/blob/5ad4179f23d9fcfe0a12574c35630d240bf0106e/wgsl/index.bs#L4472 

The prose description could be:

> An `@`, then
> the attribute name, then
> in some cases a parenthesized list of operands.
>  Each kind of attribute constrains the number and kinds of operands it can take.

Reply:
@dneto0 I thought we discussed in the editors' call that if we have general version precede and then have specific ones, it should correctly correspond to the latter, like the diagnostics special case in one of your previous PRs:

https://github.com/gpuweb/gpuweb/pull/3912

How is it factually wrong?

Reply:
And this rule is only part of `syntax/attribute` without direct use elsewhere.

Reply:
It's wrong because  in `@builtin(position)` the `position` token is not an expression.

This part of the change says that every operand to an attribute is an expression, when it is not.

> And this rule is only part of syntax/attribute without direct use elsewhere.

Right, it's not the general rule when your text makes it seem like it's the general rule.
The PR says: 
` Attributes in WGSL have the following general form...`

Regarding:
> we discussed in the editors' call that if we have general version precede and then have specific ones

What I agreed is if both the general and the specific ones are alternatives in right-hand-side of a *single* rule.

For example, [here](https://github.com/gpuweb/gpuweb/pull/3912/files#diff-2032f6927a176539ba85feb08057bf2a997c30e681544f331149121c4f52225dR26)  it has this

```
attribute:
  `@` ident_pattern_token ....
|  `@` `diagnostic` ( ...
```
and here the conflict would be between ident_pattern_token and `diagnostic`.

Reply:
@dneto0 doing in a single rule is a good way, it was actually my first iteration of this PR. I just have a small question: is it OK to do it like:

```
attribute:
  `@` ident_pattern_token ...
| diagnostic_attr
```

so that I can keep the proposed layout for subsectioning of attributes by pulling the specific rules there?

Thank you.

Reply:
I'd like to preserve the gain of reduced duplication of same definition (attribute parameters), hence, although keeping them in the right-hand-side of a single rule, it would be a great balance if we can keep the specific ones as rules with name